In [1]:
import pandas as pd

In [2]:
path = "Resources/purchase_data.csv"
file = pd.read_csv(path)
file

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male      101   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price  
0    Extraction, Quickblade Of Trembling Hands   3.53  
1                            Frenzied Scimitar   1.56  
2                                 Final Critic   4.88  
3                                  Blindscythe   3.27  
4                                         Fury   1.44  
..                                         ...    ...  
775                                       Wolf   3.54  
776                           Exiled Doomblade   1.63  
777      Celeste, Incarnation of the Corrupted   3.46  
778                               Final Critic   4.19  
779                                       Dawn   4.60  

[780 rows x 7 columns]

In [3]:
#Player Count
#Display the total number of players

Total_players = len(file["SN"].value_counts())
total_table = pd.DataFrame({"Total Players":[Total_players]})
total_table

Total Players
0            576

In [4]:
#Purchasing Analysis (Total)
#Run basic calculations to obtain number of unique items, average price, etc.
#Create a summary data frame to hold the results
#Display the summary data frame

unique_items = file["Item ID"].unique()
count_unique = len(unique_items)
avg_price = round(file["Price"].mean(),2)
avg_price = f"${avg_price:}"
num_purchases = len(file["Purchase ID"])
total_revenue = file["Price"].sum()
total_revenue = f"${total_revenue:,}"
summary1_df = pd.DataFrame({"Number of Unique items":[count_unique],
                            "Average price":[avg_price],
                            "Number of Purchases":[num_purchases],
                           "Total Revenue":[total_revenue]})
summary1_df


Number of Unique items Average price  Number of Purchases Total Revenue
0                     183         $3.05                  780     $2,379.77

In [5]:
#Gender Demographics
#Percentage and Count of Male Players
#Percentage and Count of Female Players
#Percentage and Count of Other / Non-Disclosed

file_unique = file.drop_duplicates(subset="SN",keep="first",inplace=False)
file_gender = file_unique["Gender"].value_counts()
file_male = len(file_unique.loc[file_unique["Gender"] == "Male", :])
file_male_percent = round((file_male/Total_players)*100,2)
file_male_percent = f"{file_male_percent:}%"
file_female = len(file_unique.loc[file_unique["Gender"] == "Female", :])
file_female_percent = round((file_female/Total_players)*100,2)
file_female_percent = f"{file_female_percent:}%"
file_others = len(file_unique.loc[file_unique["Gender"] == "Other / Non-Disclosed", :])
file_others_percent = round((file_others/Total_players)*100,2)
file_others_percent = f"{file_others_percent:}%"
gender_table = pd.DataFrame({"Total Count":file_gender,
                            "Percentage of Players":[file_male_percent,file_female_percent,file_others_percent]})
gender_table


Total Count Percentage of Players
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%

In [6]:
#Purchasing Analysis (Gender)
#Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender
#Create a summary data frame to hold the results
#Optional: give the displayed data cleaner formatting
#Display the summary data frame

file_gender_group = file.groupby(["Gender"])
file_gender_summary = file_gender_group["Purchase ID"].count()
file_gender_purchases = round(file_gender_group["Price"].mean(),2)
file_gender_purchases = '$' + file_gender_purchases.astype(str)
file_gender_totalpurchase = round(file_gender_group["Price"].sum(),2)
file_gender_totalpurchase = '$' + file_gender_totalpurchase.astype(str)
avg_male_purchase = file.loc[file["Gender"] == "Male", :]
avg_male_purchase = round((avg_male_purchase["Price"].sum())/file_male,2)
avg_male_purchase = f"${avg_male_purchase:}"
avg_female_purchase = file.loc[file["Gender"] == "Female", :]
avg_female_purchase = round((avg_female_purchase["Price"].sum())/file_female,2)
avg_female_purchase = f"${avg_female_purchase:}"
avg_others_purchase = file.loc[file["Gender"] == "Other / Non-Disclosed", :]
avg_others_purchase = round((avg_others_purchase["Price"].sum())/file_others,2)
avg_others_purchase = f"${avg_others_purchase:}"
file_summary_gender = pd.DataFrame({"Purchase Count":file_gender_summary,"Average Purchase Price":file_gender_purchases,
                                    "Total Purchase Value":file_gender_totalpurchase,
                                    "Avg Total Purchase per Person":[avg_female_purchase,avg_male_purchase,avg_others_purchase]})
file_summary_gender

Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                   $3.2   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Total Purchase per Person  
Gender                                                                    
Female                             $361.94                         $4.47  
Male                              $1967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

In [7]:
#Age Demographics
#Establish bins for ages
#Categorize the existing players using the age bins. Hint: use pd.cut()
#Calculate the numbers and percentages by age group
#Create a summary data frame to hold the results
#Optional: round the percentage column to two decimal points
#Display Age Demographics Table

age_max = file["Age"].max()
age_min = file["Age"].min()
bins = [0,9,14,19,24,29,34,39,50]
group_names = ("<10","10-14","15-19","20-24","25-29","30-34","35-39","40+")
file_unique["Age Range"] = pd.cut(file_unique["Age"],bins,labels=group_names)
Age_count = file_unique.groupby("Age Range")["Purchase ID"].count()
Age_percent = round((Age_count / Total_players)*100,2)
Age_percent = Age_percent.astype(str)+'%'
Age_summarytable = pd.DataFrame({"Total Count":Age_count,"Percentage of Players":Age_percent})
Age_summarytable


C:\Users\vijay\Anaconda3\lib\site-packages\ipykernel_launcher.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  del sys.path[0]


Total Count Percentage of Players
Age Range                                   
<10                 17                 2.95%
10-14               22                 3.82%
15-19              107                18.58%
20-24              258                44.79%
25-29               77                13.37%
30-34               52                 9.03%
35-39               31                 5.38%
40+                 12                 2.08%

In [8]:
#Purchasing Analysis (Age)
#Bin the purchase_data data frame by age
#Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below
#Create a summary data frame to hold the results
#Optional: give the displayed data cleaner formatting
#Display the summary data frame

file["Age Range"] = pd.cut(file["Age"],bins,labels=group_names)
range_purchase_count = file.groupby("Age Range")["Price"].count()
range_purchase_avg = round(file.groupby("Age Range")["Price"].mean(),2)
range_purchase_avg = '$' + range_purchase_avg.astype(str)
range_purchase_value = round(file.groupby("Age Range")["Price"].sum(),2)
range_purchase_value = '$'+ range_purchase_value.astype(str)
range_avg_person = round(file.groupby("Age Range")["Price"].sum()/(Age_count),2)
range_avg_person = '$' + range_avg_person.astype(str)
range_table_summary = pd.DataFrame({"Purchase Count":range_purchase_count,"Average Purchase Price":range_purchase_avg,
                                    "Total Purchase Value":range_purchase_value,
                                    "Avg Total Purchase per Person":range_avg_person})
range_table_summary

Purchase Count Average Purchase Price Total Purchase Value  \
Age Range                                                               
<10                    23                  $3.35               $77.13   
10-14                  28                  $2.96               $82.78   
15-19                 136                  $3.04              $412.89   
20-24                 365                  $3.05             $1114.06   
25-29                 101                   $2.9               $293.0   
30-34                  73                  $2.93               $214.0   
35-39                  41                   $3.6              $147.67   
40+                    13                  $2.94               $38.24   

          Avg Total Purchase per Person  
Age Range                                
<10                               $4.54  
10-14                             $3.76  
15-19                             $3.86  
20-24                             $4.32  
25-29                             $3.81  
30-34                             $4.12  
35-39                             $4.76  
40+                               $3.19

In [9]:
#Top Spenders
#Run basic calculations to obtain the results in the table below
#Create a summary data frame to hold the results
#Sort the total purchase value column in descending order
#Optional: give the displayed data cleaner formatting
#Display a preview of the summary data frame

name_purchases = file.groupby("SN")["Purchase ID"].count()
name_avg_price = file.groupby("SN")["Price"].mean()
name_total_value = file.groupby("SN")["Price"].sum()
top_spenders = pd.DataFrame({"Purchase Count":name_purchases,"Average Purchase Price":name_avg_price,
                            "Total Purchase Value":name_total_value})
top_spenders_summary = top_spenders.sort_values(["Total Purchase Value"],ascending = False)
top_spenders_summary["Total Purchase Value"] = top_spenders_summary["Total Purchase Value"].map("${:.2f}".format)
top_spenders_summary["Average Purchase Price"] = top_spenders_summary["Average Purchase Price"].map("${:.2f}".format)
top_spenders_summary.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

In [10]:
#Most Popular Items
#Retrieve the Item ID, Item Name, and Item Price columns
#Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value
#Create a summary data frame to hold the results
#Sort the purchase count column in descending order
#Optional: give the displayed data cleaner formatting
#Display a preview of the summary data frame

item_purchase = file.groupby(["Item ID","Item Name"])["Purchase ID"].count()
item_total_value = file.groupby(["Item ID","Item Name"])["Price"].sum()
item_price = file.groupby(["Item ID","Item Name"])["Price"].mean()
item_purchase_summary =  pd.DataFrame({"Purchase Count":item_purchase,"Item Price":item_price,
                                       "Total Purchase Value":item_total_value})
item_purchase_table = item_purchase_summary.sort_values(["Purchase Count"],ascending = False)
item_purchase_table["Item Price"] = item_purchase_table["Item Price"].map("${:.2f}".format)
item_purchase_table["Total Purchase Value"] = item_purchase_table["Total Purchase Value"].map("${:.2f}".format)
item_purchase_table.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
82      Nirvana                                           $4.90   
19      Pursuit, Cudgel of Necromancy                     $1.02   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
108     Extraction, Quickblade Of Trembling Hands                  $31.77  
82      Nirvana                                                    $44.10  
19      Pursuit, Cudgel of Necromancy                               $8.16

In [11]:
#Most Profitable Items
#Sort the above table by total purchase value in descending order
#Optional: give the displayed data cleaner formatting
#Display a preview of the data frame

item_purchase_table2 = item_purchase_summary.sort_values(["Total Purchase Value"],ascending = False)
item_purchase_table2["Item Price"] = item_purchase_table2["Item Price"].map("${:.2f}".format)
item_purchase_table2["Total Purchase Value"] = item_purchase_table2["Total Purchase Value"].map("${:.2f}".format)
item_purchase_table2.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
92      Final Critic                                               8   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
92      Final Critic                                      $4.88   
103     Singed Scalpel                                    $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
92      Final Critic                                               $39.04  
103     Singed Scalpel                                             $34.80

The observable trends based on the above Data are as follows

1. The Game is dominated by Male players who account for more than 3/4th of the total players.
  
2. Although the game has more male players, when it comes to purchasing items Female players tend to spend more as
   indicated by the higher average amount spent on buying items as compared to male players.
   
3. This game is more popular among young adults as almost 45% of the players in the game are between the ages of 20-24 yrs
   
4. Players in the age range of 35-39 yrs tend to purchase higher priced items in the game.